store/index.js

```diff
@@ -1,27 +1,29 @@
 import keplerGlReducer, { uiStateUpdaters } from "kepler.gl/reducers";
-import {createStore, combineReducers, applyMiddleware} from "redux";
-import {taskMiddleware} from "react-palm/tasks";
-import rootReducer from '../reducers/index'; 
-
-const customizedKeplerGlReducer = keplerGlReducer
-  .initialState({
-    uiState: {
-      // hide side panel and data input window to disallow user customize the map
-      readOnly: true,
-      currentModal: null,
-      mapLegend: {
-        show: true,
-        active: true
-      },
-    }
-  });
-
+import { createStore, combineReducers, applyMiddleware } from "redux";
+import { taskMiddleware } from "react-palm/tasks";
+import rootReducer from "../reducers/index";
+import keplerLanguageMiddleware from "./keplerLanguageMiddleware";
 
+const customizedKeplerGlReducer = keplerGlReducer.initialState({
+  uiState: {
+    // hide side panel and data input window to disallow user customize the map
+    readOnly: true,
+    currentModal: null,
+    mapLegend: {
+      show: true,
+      active: true,
+    },
+  },
+});
 
 const reducers = combineReducers({
   keplerGl: customizedKeplerGlReducer,
   root: rootReducer,
 });
 
-const store = createStore(reducers, {}, applyMiddleware(taskMiddleware));
-export default store;
\ No newline at end of file
+const store = createStore(
+  reducers,
+  {},
+  applyMiddleware(taskMiddleware, keplerLanguageMiddleware)
```

Comment:
For kepler.gl, the taskMiddleware has already been handled in  `enhanceReduxMiddleware()` (see  [here](https://github.com/keplergl/kepler.gl/blob/master/src/reducers/src/middleware.ts)).  You can do e.g.

```js
import {enhanceReduxMiddleware} from 'kepler.gl/middleware';
import logger from 'redux-logger';
...
const middlewares = enhanceReduxMiddleware([keplerLanguageMiddleware, logger]);
const store = createStore(reducers, {}, applyMiddleware(...middlewares));
```
redux.logger can help to easily debug the changes in redux state:
<img width="751" alt="Screenshot 2023-07-18 at 2 18 37 PM" src="https://github.com/GeoDaCenter/reactgeoda/assets/2423887/26c08e73-020a-41c3-b14c-22a02c6a94a5">

You can add a flag `const ENABLE_LOGGER = process.env.EnableLogger || false;` to disable/enable logger.

We used to have something like
```
// add redux devtools
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
```
to help debugging, but I prefer using redux.logger : )